editoast/src/views/path/pathfinding.rs

```diff
@@ -448,6 +453,53 @@ pub mod tests {
     use crate::views::path::pathfinding::PathfindingResult;
     use crate::views::test_app::TestAppBuilder;
 
+    #[rstest]
+    async fn pathfinding_fails_when_core_responds_with_zero_length_path() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let mut core = MockingClient::new();
+        core.stub("/v2/pathfinding/blocks")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(json!({
+                "blocks":[],
+                "routes": [],
+                "track_section_ranges": [],
+                "path_item_positions": [],
+                "length": 0,
+                "status": "success"
+            }))
+            .finish();
+        let app = TestAppBuilder::new()
+            .db_pool(db_pool.clone())
+            .core_client(core.into())
+            .build();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+
+        let request = app
+            .post(format!("/infra/{}/pathfinding/blocks", small_infra.id).as_str())
+            .json(&json!({
+                "path_items":[
+                {"trigram":"WS","secondary_code":"BV"},
+                {"trigram":"SWS","secondary_code":"BV"}
```

Comment:
Hmm, as a reviewer of the code, I was expecting 2 identical path items, but I see 2 different trigrams and I'm left wondering why do we have a zero-length path. Isn't it possible to make the test even more obvious with the following?

```suggestion
                {"trigram":"WS","secondary_code":"BV"},
                {"trigram":"WS","secondary_code":"BV"}
```

Reply:
I think part of the point here is to ensure that the endpoint correctly returns the error even if the trigrams don't match?

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/9931/commits/ed12a878549fa1b8e8199791aeb95defd7a9ba87